### DISCUSSION TITLE
NullreferenceException encountered while deserializing TMDL

### DISCUSSION THREAD

**Discussion by UlrikDeltaFact** [TabularEditor/TabularEditor3#1208] [Other Topics]
We work extensively with saving our model to folder, and using git for tracking changes, and publishing features. But since the release of 
version 3.12 + 3.13 of tabular editor we get the error you see in the screenshot - and cannot open the model. However, downgrading to version 3.11 makes everything work again.

Hence, I'm inclined to believe that some sort of bug have crept into TE3 during release of new features, that was not intended.

![image](https://github.com/TabularEditor/TabularEditor3/assets/154503566/ee4e985a-5d28-4d48-8261-a099b8a245b4)

I'm thinking that there is some null exception handling that needs to be implemented.

Regards

**Comment by otykier** [comment]
TMDL is still in preview, and breaking changes to the serializer/deserializer between different versions are unfortunately to be expected.

In general, any issues during deserialization can be worked around by using the last known working version, saving the model temporarily as a .bim file, then, using the newest version of TE3, saving it again as TMDL.

**Reply by UlrikDeltaFact** [reply]
Thanks for the answer. Inspired by your workaround we found that deleting the files in the folder and opening the model from the service  followed by saving it back to the folder again also works as a workaround. This other method is not dependent on having the old version of TE3 installed, and not need to create a BIM file.

Again, thank you for your answer.

Regards.

**Reply by mlonsk** [reply]
Yes, that is also a way of upgrading to the newest TMDL version. 
For Daniel's method you can always download the portable Tabular Editor files when you briefly want to use an earlier version of Tabular Editor. Then you do not have to install and uninstall every time.
